### PR TITLE
feat: create endpoint that return tree details

### PR DESCRIPTION
**Pull request by lfjnascimento**
create an `tree/<str:commit_hash>` endpoint that returns all builds and  a summary

To test it you can:
- run the dev server `poetry python manage.py runserver`
- Make a get request to `http://localhost:8000/api/tree/22a40d14b572deb80c0648557f4bd502d7e83826`

### CODE REVIEW

File: backend/kernelCI_app/views.py [hunk 1]
```diff
@@ -46,3 +46,22 @@ def get(self, _):
...
+            SELECT builds.*
+            FROM
+                builds
+            INNER JOIN
+                checkouts ON checkouts.id = builds.checkout_id
+            WHERE checkouts.git_commit_hash = %s;
+            """,
+            [commit_hash]
+        )
+        print('commit_hash:', commit_hash)
```
Comment (mari1912) [inline backend/kernelCI_app/views.py:65]: do we need this print?

**Review by mari1912 — APPROVED** [review]
LGTM

**Review by viniciuslacerda0 — APPROVED** [review]

File: backend/kernelCI_app/serializers.py [hunk 2]
```diff
@@ -67,3 +67,14 @@ def get_test_status(self, obj):
...
+
+class TreeDetailsSerializer(serializers.Serializer):
+    id = serializers.CharField()
+    architecture = serializers.CharField()
+    config_name = serializers.CharField()
+    valid = serializers.BooleanField()
+    start_time = serializers.CharField()
+    duration = serializers.CharField()
+    compiler = serializers.CharField()
+    test_status = serializers.CharField()
```
Comment (mari1912) [inline backend/kernelCI_app/serializers.py:80]: we will also need the link of logs

Comment (mari1912) [reply backend/kernelCI_app/serializers.py:80]: and config url

**Review by mari1912 — COMMENTED** [review]
I'm not sure test_status is working properly, it returns 0 for all of them when I tested it for http://localhost:8000/api/tree/189ee9735a4b2e8095b1a6c088ebc8e133872471 for example. When I search only `localhost:8000/api/tree` it gets some tests for this commit hash. Here I'm using kcidb to test. It may be a good idea to change it to default

File: backend/kernelCI_app/serializers.py (hunk 2 above)
Comment (lfjnascimento) [reply backend/kernelCI_app/serializers.py:80]: You can test if there is a `test.status` for a specific build by runing:

```SQL
SELECT tests.status
FROM
  tests
WHERE tests.build_id = <build_id>
```

You'll notice that  most of the build you test there a test

**Review by mari1912 — APPROVED** [review]

**Pull request merged**